consumer/src/tree.rs

```diff
@@ -184,9 +184,9 @@ impl State {
     fn serialize(&self) -> TreeUpdate {
         let mut nodes = Vec::new();
 
-        fn traverse(state: &State, nodes: &mut Vec<(NodeId, NodeData)>, id: NodeId) {
+        fn traverse(state: &State, nodes: &mut Vec<(NodeId, Arc<NodeData>)>, id: NodeId) {
             let node = state.nodes.get(&id).unwrap();
-            nodes.push((id, (*node.data).clone()));
+            nodes.push((id, node.data.clone()));
```

Comment:
I think I would have used `Arc::clone(&node.data)` here to make it clear that it's not a full clone, but that's only a personal preference I guess.